docs/sdks/uid2-sdk-ref-java.md

```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
 
 - [https://central.sonatype.com/artifact/com.uid2/uid2-client](https://central.sonatype.com/artifact/com.uid2/uid2-client)
 
-## Usage for DSPs
+## Initialization
+
+The initialization step depends on the role, as shown in the following table.
+
+| Role        | Create Instance of Class | Link to Instructions                              |
+|:------------| :--- |:--------------------------------------------------|
+| `DSP`       | BidstreamClient | [Usage for DSPs](#usage-for-dsps)                 |
+| `Sharer`    | SharingClient | [Usage for UID2 Sharers](#usage-for-uid2-sharers) |
+| `Publisher` | Uid2PublisherClient | [Usage for Publishers](#usage-for-publishers)     |
 
-The initialization function configures the parameters necessary for the SDK to authenticate with the UID2 service. It also allows you to configure retry intervals in the event of errors.
+You will need to provide the values necessary for the SDK to authenticate with the UID2 service.
 
-| Parameter | Description | Recommended Value |
-| :--- | :--- | :--- |
-| `endpoint` | The endpoint for the UID2 service. | N/A |
-| `authKey` | The authentication token that belongs to the client. For access to UID2, see [Contact Info](../getting-started/gs-account-setup.md#contact-info). | N/A |
+| Parameter | Description | 
+| :--- | :--- |
+| `endpoint` | The endpoint for the UID2 service. See [Environments](../getting-started/gs-environments) | 
+| `auth_key` | The API key. See [UID2 Credentials](../getting-started/gs-credentials). | 
+| `secret_key` | The client secret. See [UID2 Credentials](../getting-started/gs-credentials). | 
 
 ### Interface 
 
-The interface allows you to decrypt UID2 advertising tokens and return the corresponding raw UID2. 
+The `BidstreamClient` class allows you to decrypt UID2 tokens into raw UID2s.
+For details on the bidding logic for handling user opt-outs, see [DSP Integration Guide](../guides/dsp-guide.md).
+
+The `SharingClient` class allows you to encrypt raw UID2s into UID2 tokens and decrypt UID2 tokens into raw UID2s.
 
 :::note
 When you use an SDK, you do not need to store or manage decryption keys.
 :::
 
-If you're a DSP, for bidding, call the interface to decrypt a UID2 advertising token and return the UID2. For details on the bidding logic for handling user opt-outs, see [DSP Integration Guide](../guides/dsp-guide.md).
+### Encryption Response Content
+
+When encrypting with the `SharingClient`, the SDK returns the information shown in the following table.
+
+| Property         | Description                                                                                                                                     |
```

Comment:
Same comment as below, we have methods not properties

Reply:
Yeah, originally I thought that was only comments. I should make that more accurate.